src/riak_kv_ts_svc.erl

```diff
@@ -804,132 +581,130 @@ make_rpberrresp(Code, Message) ->
+make_failed_listkeys_resp(Reason) ->
     make_rpberrresp(
       ?E_LISTKEYS,
       flat_format("Failed to list keys: ~p", [Reason])).
 
-table_create_fail_response(Table, Reason) ->
+make_table_create_fail_resp(Table, Reason) ->
     make_rpberrresp(
       ?E_CREATE, flat_format("Failed to create table ~s: ~p", [Table, Reason])).
 
-table_activate_fail_response(Table) ->
+make_table_activate_fail_resp(Table) ->
     make_rpberrresp(
       ?E_ACTIVATE,
       flat_format("Failed to activate table ~s", [Table])).
 
-table_not_activated_response(Table) ->
+make_table_not_activated_resp(Table) ->
     make_rpberrresp(
       ?E_TABLE_INACTIVE,
-      flat_format("~ts is not an active table.", [Table])).
+      flat_format("~ts is not an active table", [Table])).
 
-table_created_missing_response(Table) ->
+make_table_created_missing_resp(Table) ->
     make_rpberrresp(
       ?E_CREATED_GHOST,
       flat_format("Table ~s has been created but found missing", [Table])).
 
 to_string(X) ->
     flat_format("~p", [X]).
 
-%% Returns a tuple with a list of request IDs and an error tally
-invoke_async_put(BuildRObjFun, AsyncPutFun, _BatchPutFun, {individual, Records}) ->
-    lists:map(fun(Record) ->
-                      {LK, RObj} = BuildRObjFun(Record),
-                      {ok, ReqId} = AsyncPutFun(RObj, LK),
-                      ReqId
-                end,
-              Records);
-invoke_async_put(BuildRObjFun, _AsyncPutFun, BatchPutFun, {batches, Batches}) ->
-    lists:map(fun(Batch) ->
-                      RObjs = lists:map(BuildRObjFun, Batch),
-                      {ok, ReqId} = BatchPutFun(RObjs),
-                      ReqId
-                end,
-              Batches).
 
 %% helpers to make various responses
 
--spec make_tsqueryresp([] | {[riak_pb_ts_codec:tscolumnname()],
-                             [riak_pb_ts_codec:tscolumntype()],
-                             [[riak_pb_ts_codec:ldbvalue()]]}) -> ts_query_response().
-make_tsqueryresp({_, _, []}) ->
+make_tsgetresp(ColumnNames, ColumnTypes, Rows) ->
+    {tsgetresp, {ColumnNames, ColumnTypes, Rows}}.
+
+make_tsqueryresp({_ColumnNames, _ColumnTypes, []}) ->
+    %% tests (and probably docs, too) expect an empty result set to be
+    %% returned as {[], []} (with column names omitted). Before the
+    %% TTB merge, the columns info was dropped in
+    %% riak_kv_pb_ts:encode_response({reply, {tsqueryresp, Data}, _}).
+    %% Now, because the TTB encoder is dumb, leaving the special case
+    %% treatment in the PB-specific riak_kv_pb_ts lets empty query
+    %% results going through TTB still have the columns info, which is
+    %% not what the caller might expect. We ensure uniform {[], []}
+    %% for bot PB and TTB by preparing this term in advance, here.
     {tsqueryresp, {[], [], []}};
-make_tsqueryresp({ColumnNames, ColumnTypes, Rows}) ->
-    {tsqueryresp, {ColumnNames, ColumnTypes, Rows}}.
+make_tsqueryresp(Data = {_ColumnNames, _ColumnTypes, _Rows}) ->
```

Comment:
The process of producing `ColumnNames` and `ColumnTypes` is different between `riak_kv_ts_api:query` and `:get_data`. For the not-found case to have uniform `{[], []}` presented to the client, columns info needs to be specially stripped from the query return values, while get_data already returns none. Hence this difference in `make_ts{query,get}resp`.